### ISSUE TITLE
Some Stations Do Not Appear, Uncompletable Campaign

### ISSUE THREAD

**Issue by KevUndead** [Regalis11/Barotrauma#3537]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
When on a mission, the next station we travel to may sometimes not load, and therefore either can't be completed (like for delivery missions) or, if it's a hunting mission, means we have to backtrack all the way to the original station we departed from. On one instance we were actually completely landlocked; our original station was in a corner, and BOTH of the next stations were bugged and un-dockable.

**Steps To Reproduce**
Random. Start a Multiplayer Campaign mode game, get a mission and begin it, and hope the bug happens. It should, since it happened more often than not.

**Version**
v0.10.4.0 , running on Manjaro (Arch Linux distro).

**Additional information**
Don't know if this matters or not: the sub we used was the Dugong, difficulty was 20%, 5 bots set to Fill mode, all subs purchasable, and again it's Campaign on Multiplayer.

**Comment by mygamingaccount** [comment]
Is your issue related to https://github.com/Regalis11/Barotrauma/issues/3470?

**Comment by Kaillera** [comment]
There's a flashy button at the top center of the screen for any host/admin/mod to "Enter XXX" at the end of the level.

**Comment by KevUndead** [comment]
> Is your issue related to #3470?

I believe so, didn't find that issue when I searched before. Never had a formations mission, but I'm sure it's a similar situation.


> There's a flashy button at the top center of the screen for any host/admin/mod to "Enter XXX" at the end of the level.

That button does absolutely nothing to help. The map will come up but nothing will be available to enter.

**Comment by mygamingaccount** [comment]
You select your next destination on the map to continue the campaign

**Comment by KevUndead** [comment]
> You select your next destination on the map to continue the campaign

Like I said before, that does not work.

**Comment by KevUndead** [comment]
A quick update cause I'm dumb and forget things: the bug is prevalent in multiplayer, and according to my friend it's _bypassable_ in singleplayer like @mygamingaccount suggested, but the initial station you _were_ going to for your current mission will still be bugged out and missing.

**Comment by Regalis11** [comment]
To be clear, normally the steps to proceed from a natural formation (i.e. any location with no outpost) are as follows:
1. Open the map by clicking the blinking "Enter xxx" button on top of the screen
2. Click on a destination
3. Click on the "Start" button in the panel that opens

Do note that in multiplayer only the host/admins or players with campaign management permissions are allowed to select the destination and start the new round. That could be an explanation for why you can't end the round, are you hosting the server yourself or playing on someone elses server?

**Comment by KevUndead** [comment]
I was host every time, and I would attempt those steps, but step 3 would not be possible. The destination's panel would show up along with its description, but no other button in or around it, causing us to have to forfeit the mission and return to the previous outpost.

I would set up the server for 2-3 players depending on which of my friends joins me, set a password, pick our characters, load our campaign, and choose a mission while we're on the starting outpost. From then on, it's a dice roll: either the next station is there and our missions can continue, or they're not, and odds tended to favor _not_ more often. When that happens we decide to try going back and pathing to a different outpost, but again since the odds were always in not spawning an outpost we would eventually reach what is essentially a land-lock, unable to progress any further in the campaign.

In those more recent times when I decided to make a bug report, we got land-locked almost instantly, being stuck in either a corner or edge outpost on the map, with all connecting ones missing. Trying to do the 3 steps you mentioned leads to that 3rd step not existing.
